version: "3.9"

services:
  traefik:
    image: traefik:v3.1
    command:
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--entrypoints.web.address=:80"
    ports:
      - "8081:80"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  api1:
    image: nginx
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.api.rule=Host(pikachu.localhost) && PathPrefix(/)"
      - "traefik.http.routers.api.entrypoints=web"
      - "traefik.http.services.api.loadbalancer.server.port=80"

  api2:
    image: nginx
    labels:
      - "traefik.enable=true"
      - "traefik.http.services.api.loadbalancer.server.port=80"

  api-legendario:
    image: nginx
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.legendario.rule=Host(pikachu.localhost) && PathPrefix(/legendario)"
      - "traefik.http.routers.legendario.entrypoints=web"
      - "traefik.http.services.legendario.loadbalancer.server.port=80"
      - "traefik.http.routers.legendario.middlewares=legendario-logger"
      - "traefik.http.middlewares.legendario-logger.plugin.traefik-logger.loglevel=INFO"
